app/views/trains/show.html.erb

```diff
@@ -25,5 +25,28 @@
   <% end %>
 </p>
 
+<p>
+  <strong>Carriages</strong>
+  <p>
+    <% count_carraiage_types(@train).each do |key, value| %>
+      <%=  "#{key.capitalize} - #{value}" %>
+    <% end %>
+  </p>
+  <p>      
+    <strong>Amount places:</strong>
+    <ul>
+      <% places = count_carraiage_places(@train) %>
```

Comment:
Please don't use calculations on the view side

Reply:
If do it in controller?

Reply:
There is [another way](https://nithinbekal.com/posts/rails-presenters/), but if you need to operate a data from the database - you can do it in a model/